test/t8_geometry/t8_geometry_implementations/t8_gtest_geometry_cad.cxx

```diff
@@ -365,6 +367,185 @@ TEST (t8_gtest_geometry_cad, jacobian)
 }
 #endif /* T8_WITH_OCC */
 
+#if T8_WITH_OCC
+/* The test checks if the mapping algorithms for curved 2d elements do not shift values on an edge which is not curved.
+ * In that case, the cad geometry should output the same out_coords as the linear geometry function. */
+class class_2d_element_linear_cad_curve: public testing::TestWithParam<t8_eclass_t> {
+ protected:
+  void
+  SetUp () override
+  {
+    eclass = GetParam ();
+    T8_ASSERT (0 <= eclass && eclass < T8_ECLASS_COUNT);
+    Handle_Geom_Curve cad_curve;
+    TColgp_Array1OfPnt point_array (1, 3);
+
+    /*  x--> u-parameter
+    *   
+    *                 curve
+    *  ----------------------------------
+    * 
+    *  0               0.5              1
+    */
+
+    point_array (1) = gp_Pnt (0.0, 0.0, 0.0);
+    point_array (2) = gp_Pnt (0.5, 0.0, 0.0);
+    point_array (3) = gp_Pnt (1.0, 0.0, 0.0);
+
+    cad_curve = GeomAPI_PointsToBSpline (point_array).Curve ();
+    shape = BRepBuilderAPI_MakeEdge (cad_curve).Edge ();
+  }
+
+  void
+  TearDown () override
+  {
+    /* The cmesh is destroyed in the test itself. */
+  }
+  t8_cmesh_t cmesh;
+  t8_eclass_t eclass;
+  TopoDS_Shape shape;
+  /* The arrays prescribe the linkage of the element. Edge 2 of the element is linked and the face is unlinked. */
+  int faces[1] = { 0 };
+  int edges[8] = { 0, 0, 1, 0, 0, 0, 0, 0 };
```

Comment:
When we change the orientation, we should also link different edges. Why doesn't this test fail when we link the cad edge to an element edge, which does not correspond with it?